ps3t_game_list.php

```diff
@@ -96,7 +96,14 @@
         
         $nextpage = 1;
         
-        for ($j = 1; $j <= 9; $j++) {
+        // Set number of pages dynamically
+		$init_html = file_get_html('http://www.' . $siteurl . '/browsegames/' . $gametype[$i] .
+										  '/' . $gamealpha . '/');
+		$table_object = $init_html->find('div.bl_la_main div.divtext table', 1);
+		$page_object = $table_object->find('td', 0);
+		$page_num = explode(" ", $page_object)[4];

```

Comment:
This works for PST but not for XBA... their pagination links are different: https://www.xboxachievements.com/browsegames/xbox-one/t/1

You'd have to do an if statement based on `$gamesite` to handle.

Reply:
Also you should do a clear/unset for `$init_html` before you get into the for loop to free up memory (like at the bottom of the file for the `$html` variable).

Reply:
Is pagination actually different between the two?

https://www.playstationtrophies.org/browsegames/ps4/s/2
https://www.xboxachievements.com/browsegames/xbox-one/s/2

Based on the variables, it should work just fine for either... no?

Reply:
Ooooooooh oh oh oh I see what you mean now... XBA has the selectable numbers while PST has a "next page" option only. I'll see what I can do about that.

Reply:
Okay, so I also corrected the pagination. It now pulls the last page based on either PST or XBA